src/components/modal/index.tsx

```diff
@@ -0,0 +1,100 @@
+
+import { TextField } from '@mui/material';
+import { useContext, useState, useEffect } from 'react';
+import { Modal } from 'react-bootstrap';
+import Button from '@mui/material/Button';
+import { IRepositoryProps } from '../../types/repositories';
+import { UserProfileContext } from '../../context/user';
+import { getProfileUser } from '../../service';
+import { IProfileUserProps } from '../../types/profileUser';
+
+export const ModalComponent = ({children}: any) => {
+  
+  const [show, setShow] = useState(false);
+  const [nickname, setNickName] = useState('');
+  const [repositories, setRepositories] = useState<IRepositoryProps[]>([]);
+  const { setProfileUserLocalStored } = useContext(UserProfileContext);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  const handleSetNickname = async () => {
+    if (nickname !== '') {
+      
+      var resProfile = await getProfileUser(nickname);
```

Comment:
Don't use `var` key to declare variable into javascript. It has a little problem with scope. The var key makes the variable overflow from the its scope to another. So it's recommended by community just use `let` key or `const` if the variable doesn't receive changes